plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -372,7 +374,8 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
          auto bsf = chain.create_block_state_future( block );
 
          // abort the pending block
-         chain.abort_block();
+         vector<transaction_metadata_ptr> aborted_trxs = chain.abort_block();
+         _unapplied_transactions.add_aborted( std::move( aborted_trxs ) );
```

Comment:
I would move this line to after the call to `add_forked` below.

Reply:
Done